azimuth_caas_operator/utils/ansible_runner.py

```diff
@@ -80,14 +81,48 @@ async def ensure_deploy_key_secret(client, cluster: cluster_crd.Cluster):
     return base64.b64decode(secret.data["id_ed25519.pub"]).decode()
 
 
+async def get_global_extravars(client):
+    """
+    Retrieves the global extra vars from the specified secret.
+    """
+    # The secret is specified in the form namespace/name
+    secret_info = os.environ.get("GLOBAL_EXTRAVARS_SECRET")
+    if not secret_info:
+        return {}
+    secret_resource = await client.api("v1").resource("secrets")
+    secret_namespace, secret_name = secret_info.split("/", maxsplit=1)
+    try:
+        secret = await secret_resource.fetch(secret_name, namespace=secret_namespace)
+    except ApiError as exc:
+        if exc.status_code == 404:
+            # If the secret does not exist, log it but proceed with no global extravars
+            LOG.warn("global extravars secret was specified but does not exist")
+            return {}
+        else:
+            raise
+    # We parse each value as YAML and merge them together
+    # If the value doesn't parse as YAML, log it but proceed as if the key didn't exist
+    global_extravars = {}
+    for key, b64data in sorted(secret.get("data", {}).items()):
+        data = base64.b64decode(b64data)
+        try:
+            global_extravars.update(yaml.safe_load(data))
+        except yaml.YAMLError:
+            LOG.exception("global extravars key is not valid YAML - %s", key)
```

Comment:
I don't like how is is quite a silent failure here, should we throw and exception here? We don't want to accidentally create a platform that is missing these for some obscure reason.